vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSearchController.java

```diff
@@ -0,0 +1,124 @@
+package edu.asu.diging.vspace.web.staff;
+
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import edu.asu.diging.vspace.core.model.impl.Module;
+import edu.asu.diging.vspace.core.model.impl.Slide;
+import edu.asu.diging.vspace.core.model.impl.Space;
+import edu.asu.diging.vspace.core.services.IStaffSearchManager;
+
+@Controller
+public class StaffSearchController {
+
+    @Autowired
+    private IStaffSearchManager staffSearchManager;
+
+    @RequestMapping(value = "/staff/search")
```

Comment:
at least for the pagination, there should individual asynchronous calls be made. No reason to search all types again, if only spaces are paginated through. It might make sense to make all search request individual requests or just the paginating ones.